src/backend/winit/mod.rs

```diff
@@ -116,12 +127,13 @@ pub struct WinitEventLoop {
     resize_notification: Rc<Cell<Option<Size<i32, Physical>>>>,
     /// Whether winit is using Wayland or X11 as it's backend.
     is_x11: bool,
+    backend: Rc<RefCell<WinitGraphicsBackend>>,
 }
 
 /// Create a new [`WinitGraphicsBackend`], which implements the
 /// [`Renderer`](crate::backend::renderer::Renderer) trait and a corresponding
 /// [`WinitEventLoop`].
-pub fn init<L>(logger: L) -> Result<(WinitGraphicsBackend, WinitEventLoop), Error>
+pub fn init<L>(logger: L) -> Result<(Rc<RefCell<WinitGraphicsBackend>>, WinitEventLoop), Error>
```

Comment:
The surface is already stored in a type with interior mutability, is this change needed?

Reply:
I think so? Um... months have passed by, but if I recall correctly, I wrote some messages on Matrix on this exact issue.

Wait.. let me check again..

Reply:
![Smithay master branch documentation showing the Bind<Rc<EGLSurface>> trait implemented to Gles2Renderer](https://user-images.githubusercontent.com/66000635/190857717-fc4671fc-46d5-4e62-b789-2c45bce09ae6.png)

Since the surface may be null (Android EGL surfaces are only present when the app is active), we have to store it as an Option, right? But obviously compiler issues will not allow us to unwrap an Option from Rc<Option<EGLSurface>>.